src/module.ts

```diff
@@ -110,6 +111,9 @@ export default defineNuxtModule<ModuleOptions>({
     }
 
     if (nuxt.options.security.xssValidator) {
+      if (nuxt.options.security.xssValidator.methods) {
+        xssMethods.push(...nuxt.options.security.xssValidator.methods)
```

Comment:
issue: there is a better way to do it ;)

Check out how AllowedMethodsRestricter was implemented and please refactor your solution to it. It is simpler and utilizes the runtimeConfig and routeRules

Reply:
Sure thing!